### PR TITLE
article history, adds pre-print events

### PR DESCRIPTION
**Pull request by lsh-0**

### CODE REVIEW

File: src/model/article-history.v1.yaml [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
+          - received
+    required:
+        - versions
+-
+    type: object
+    properties:
+        type:
+            type: string
+            enum:
+                - preprint
```
Comment (lsh-0) [inline src/model/article-history.v1.yaml:34]: > The content-type/date-type values we are going to use are:
preprint
accepted-manuscript
version-of-record
corrected-version-of-record

but not in this round of changes

**Comment by nlisgo** [comment]
We certainly should add an example of the history with the preprint property present in the samples.

**Review by gnott — APPROVED** [review]

**Comment by gnott** [comment]
It looks good to me, with the assumption this doesn't need any version bumps because it will be backwards compatible. I'm curious though whether you think it will affect the data returned by the API `articles/xxxxx/versions` endpoint, and if so I might want to check it wouldn't have unexpected integration with `elife-bot` logic.

**Comment by lsh-0** [comment]
I don't think this is going to work at all. I think I misunderstood [anyOf](https://json-schema.org/understanding-json-schema/reference/combining.html) (or something). This is `dist/samples/article-history/v1/multiple-versions.json`:

```json
{
    "received": "2015-12-29",
    "accepted": "2016-03-27",
    "versions": [
        {
            "status": "vor",
            "id": "14107",
            "version": 3,
            "type": "research-article",
            "doi": "10.7554/eLife.14107",
            "authorLine": "Yongjian Huang et al.",
            "title": "Molecular basis for multimerization in the activation of the epidermal growth factor",
            "stage": "published",
            "published": "2016-03-28T00:00:00Z",
            "versionDate": "2016-04-30T00:00:00Z",
            "statusDate": "2016-04-30T00:00:00Z",
            "volume": 5,
            "elocationId": "e14107",
            "pdf": "https://elifesciences.org/content/4/e14107.pdf",
            "subjects": [
                {
                    "id": "genomics-evolutionary-biology",
                    "name": "Genomics and Evolutionary Biology"
                }
            ],
            "impactStatement": "A new hominin species has been unearthed in the Dinaledi Chamber of the Rising Star cave system in the largest assemblage of a single species of hominins yet discovered in Africa.",
            "image": {
                "thumbnail": {
                    "uri": "https://iiif.elifesciences.org/lax/09560%2Felife-09560-fig1-v1.tif",
                    "alt": "",
                    "attribution": [
                        "By some person."
                    ],
                    "source": {
                        "mediaType": "image/jpeg",
                        "uri": "https://iiif.elifesciences.org/lax/09560%2Felife-09560-fig1-v1.tif/full/full/0/default.jpg",
                        "filename": "an-image.jpg"
                    },
                    "size": {
                        "width": 4194,
                        "height": 4714
                    },
                    "focalPoint": {
                        "x": 25,
                        "y": 75
                    }
                }
            }
        },
        {
            "status": "poa",
            "id": "14107",
            "version": 2,
            "type": "research-article",
            "doi": "10.7554/eLife.14107",
            "authorLine": "Yongjian Huang et al.",
            "title": "Molecular basis for multimerization in the activation of the epidermal growth factor",
            "stage": "published",
            "published": "2016-03-28T00:00:00Z",
            "versionDate": "2016-03-29T00:00:00Z",
            "statusDate": "2016-03-28T00:00:00Z",
            "volume": 5,
            "elocationId": "e14107"
        },
        {
            "status": "poa",
            "id": "14107",
            "version": 1,
            "type": "research-article",
            "doi": "10.7554/eLife.14107",
            "authorLine": "Yongjian Huang et al.",
            "title": "Molecular basis for multimerization in the activation of the epidermal growth factor",
            "stage": "published",
            "published": "2016-03-28T00:00:00Z",
            "versionDate": "2016-03-28T00:00:00Z",
            "statusDate": "2016-03-28T00:00:00Z",
            "volume": 5,
            "elocationId": "e14107"
        }
    ]
}
```

My proposed spec change would have either the above *or* the pre-print, when what we really want is the pre-print within the `versions` list. It needs a `snippets/article-preprint.v1.yaml` type snippet then. (I guess?).

I'll open another PR for that in case I'm completely wrong and mis-mis-understood json-schema and this is what we actually want.

**Comment by lsh-0** [comment]
replacement PR: https://github.com/elifesciences/api-raml/pull/269

**Pull request closed without merging**